import_ldraw.py

```diff
@@ -928,6 +970,41 @@ def getColorValue(line, value):
     if value in line:
         n = line.index(value)
         return line[n + 1]
+        
+def linkedParts():
+    """Clean-up design by linking identical parts (mesh/color)"""
+    colors = []
+    for color in bpy.data.materials:
+        colors.append(color.name)
+    
+    parts = []
+    for part in bpy.data.objects:
+        if part.type == 'MESH' and part.name.split('.')[0] not in parts:
+            parts.append(part.name.split('.')[0])
+
+    for part in parts:
+        for color in colors:
+            replaceParts(part, color)
+
+def replaceParts(part, color):
+    """Replace identical meshes of part/color-combination with a linked version"""   
+    scene = bpy.context.scene
+    mat = bpy.data.materials[color]
+    mesh = None
+
+    for ob in scene.objects:
+        if ob.type == 'MESH' and ob.name.split('.')[0] == part:
+            for slot in ob.material_slots:
+                if slot.material == mat:
+                    if mesh == None:
```

Comment:
Ok, the reasoning behind the introduction of the levels is 100% linked to linked bricks. Here we go. ;-)

The basic idea behind linked bricks is to find the top-level brick and collapse all subparts/primitives into it. The orientation and position of this top-level brick is kept separately in order to add this info to Blender in 'matrix_world'. 

The next step is to find all identical top-level bricks based upon same mesh and colour. I would have preferred to do the linking based upon the mesh only, but it's a choice of Blender to connect the materials to the mesh instead of the object (because you can in doing so add different materials to different parts of the mesh, but that is not important for now).

If you find two bricks with the same colour and mesh, you remove the mesh of one of them and replace it by a link to the mesh of the other one (in reality this happens differently, but for explaining the principle this is sufficient).

In the Blender Outliner you can see that each brick-object has now the official name of the brick. The mesh has the same name with the colour added. But two equal bricks will link to the same mesh. And that is exactly the point.

Now it becomes tricky.

How can we find the top-level brick. Simple answer: check the header of the dat-files for the "Part" definition. (And later extend it with Non-official top-levels... but that is another discussion)

That works fine... unless that part-definition refers to another part-definition. This is for example the case for LEGO Minifigs Arms. The left arm refers to the right arm and mirrors it. This means that a top-level "Part" refers to a lower-level "Part".

If you only take into the "Part" definition for defining top-level bricks you will end up with an empty left arm and two right arms. I have added my test file to GitHub, so you can check this for yourself.

Solution: Introduce a level counter and only consider the bricks at level 1. This way the left arm is recognized as top-level and linked and mirrored right arm is collapsed into the left arm. As it should be.

By introducing the level counter I am convinced that I can eliminate the "self.orientation" parameter too (one that I introduced before) and make the code a bit more understandable. I am currently experimenting with this.

Hope this explains a bit my motivation for this branch (and my headaches...).